### PR TITLE
Separate pager from abort handling

### PR DESCRIPTION
**Pull request by jenswi-linaro**
Refactoring abort handling to make it easier to integrate the coming hard-float support which needs special treatment in aborts.

The commits are intended to be reviewed one by one.

### CODE REVIEW

File: core/arch/arm/kernel/abort.c [hunk 1]
```diff
@@ -0,0 +1,366 @@
...
+	     abort_type_to_str(ai->abort_type),
+	     ai->va, ai->fault_descr, ai->pc, read_ttbr0_el1(),
+	     read_contextidr_el1(),
+	     read_mpidr_el1(), (uint32_t)ai->regs->spsr);
+#endif /*ARM64*/
+#endif /*TRACE_LEVEL >= TRACE_DEBUG*/
+}
+
+#ifdef ARM32
+static void set_abort_info(uint32_t abort_type, struct thread_abort_regs *regs,
```
Comment (pascal-brand-st-dev) [inline core/arch/arm/kernel/abort.c]: May make sense to have also `abort_a32.c` and `abort_a64.c` to contain the functions that are different?

**Comment by pascal-brand-st-dev** [comment]
With or without my comment:

    Reviewed-by: Pascal Brand <pascal.brand@linaro.org>

**Comment by jforissier** [comment]
When my two comments are addressed:
`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

File: core/arch/arm/kernel/abort.c (hunk 1 above)
Comment (jenswi-linaro) [inline core/arch/arm/kernel/abort.c]: I think that would make it harder to get an overview. Perhaps I should try to minimize the number of ifdefs in this file instead?

**Comment by jenswi-linaro** [comment]
Jeromes comments addressed, tags applied and rebased.
Pascal, I'd rather address your comment later in a separate commit.

**Pull request merged**